### PR TITLE
Finished service tests

### PR DESCRIPTION
**Pull request by FairyTalePol**
Finished tests for statistics service, also added tests for Public and Private assets. Some minor fixes elsewhere

### CODE REVIEW

File: src/main/kotlin/com/example/springrestuser/controllers/AssetController.kt [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import com.example.springrestuser.`do`.AssetDO
+import com.example.springrestuser.dto.AssetDTO
+import com.example.springrestuser.services.AssetService
+import org.springframework.web.bind.annotation.DeleteMapping
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.PathVariable
+import org.springframework.web.bind.annotation.RestController
+
+@RestController
+class AssetController(
```
Comment (sn3jkeee) [inline src/main/kotlin/com/example/springrestuser/controllers/AssetController.kt:12]: Title says "Finished service tests". But here is Controller. How come?

**Review by sn3jkeee — CHANGES_REQUESTED** [review]

File: src/main/kotlin/com/example/springrestuser/controllers/AssetController.kt (hunk 1 above)
Comment (FairyTalePol) [reply src/main/kotlin/com/example/springrestuser/controllers/AssetController.kt]: Forgot to remove this file. Fixed now

File: src/test/kotlin/com/example/springrestuser/services/AssetServiceTest.kt [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    @AfterEach
+    fun afterEach() {
+        confirmVerified(storage)
+    }
+
+    @Test
+    fun `get asset successfully`() {
+        //Given
+        val expected = AssetDO(15.0, "Description", AssetType.PUBLIC)
+        expected.id = 0;
```
Comment (sn3jkeee) [inline src/test/kotlin/com/example/springrestuser/services/AssetServiceTest.kt:32]: wtf

File: src/test/kotlin/com/example/springrestuser/services/AssetServiceTest.kt [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+        //When
+        every { storage.getAsset(0) } returns expected
+        val actual = service.get(0)
+
+        //Then
+        verify(exactly = 1) { storage.getAsset(0) }
+        Assertions.assertEquals(expected, actual)
```
Comment (sn3jkeee) [inline src/test/kotlin/com/example/springrestuser/services/AssetServiceTest.kt:40]: static imporrts

**Review by sn3jkeee — CHANGES_REQUESTED** [review]

**Review by sn3jkeee — CHANGES_REQUESTED** [review]

File: src/main/kotlin/com/example/springrestuser/services/StatisticsService.kt [hunk 4]
```diff
@@ -35,11 +34,13 @@ class StatisticsService(
         if (!userStorage.isExists(userId)) {
             throw UserNotFoundException(userId)
         }
-        val assets = userAssetRelationStorage.getUserAssetRelationsByUser(userId, AssetType.PRIVATE)
+        val assets = userAssetRelationStorage.getUserAssetRelationsByUser(userId)
 
         var sum = 0.0
         for (asset in assets) {
-            sum+=asset.quantity*asset.asset.price
+            if (asset.asset.type==AssetType.PRIVATE){
```
Comment (sn3jkeee) [inline src/main/kotlin/com/example/springrestuser/services/StatisticsService.kt:41]: spaces

File: src/test/kotlin/com/example/springrestuser/services/AssetServiceTest.kt [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+        verify(exactly = 1) { storage.getAsset(0) }
+        assertEquals(expected, actual)
+    }
+
+    @Test
+    fun `get all assets successfully`() {
+        //Given
+        val expected1 = AssetDO(15.0, "Description 1", AssetType.PUBLIC, 0)
+        val expected2 = AssetDO(15.0, "Description 2 ", AssetType.PUBLIC, 1)
+        val expected3 = AssetDO(15.0, "Description 3", AssetType.PUBLIC, 2)
```
Comment (sn3jkeee) [inline src/test/kotlin/com/example/springrestuser/services/AssetServiceTest.kt:46]: better to add Private asset to the list

File: src/test/kotlin/com/example/springrestuser/services/StatisticsServiceTest.kt [hunk 6]
```diff
@@ -0,0 +1,193 @@
...
+        verify(exactly = 1) { groupAssetRelationStorage.getGroupsIdByAsset(0) }
+        verify(exactly = 1) { groupAssetRelationStorage.getGroupsIdByAsset(1) }
+    }
+
+    @Test
+    fun `get summary price of assets by user id`() {
+        //Given
+        val asset1 = AssetDO(15.0, "", AssetType.PRIVATE, 0)
+        val asset2 = AssetDO(20.0, "", AssetType.PRIVATE, 1)
+        val asset3 = AssetDO(30.0, "", AssetType.PRIVATE, 2)
```
Comment (sn3jkeee) [inline src/test/kotlin/com/example/springrestuser/services/StatisticsServiceTest.kt:79]: filtering by asset type is not tested

**Review by sn3jkeee — CHANGES_REQUESTED** [review]

**Review by sn3jkeee — APPROVED** [review]

**Pull request merged**